Lib/test/test_subprocess.py

```diff
@@ -2814,6 +2814,11 @@ def test_stopped(self):
 
         self.assertEqual(returncode, -3)
 
+    def test_kill_on_error(self):
```

Comment:
Looks it only can add positive testcase due to the core code just only 'yield self'
If somebody have a good idea about negatigve testcase, pls ping me;)

Reply:
you could create a Popen subclass that overrides either `__enter__` or `__exit__` to raise an exception.  (or do the equivalent using unittest.mock)

Reply:
Hi, Gregory, thank for your elaborative review:).
Both of them can not work in here. for the reason:
1) We use contextmanager in the new function and it can not work with `__enter__` or `__exit__`.
2) Looks like `yield proc` can not be mocked to raise a error.